implementation-guide/Implementation Guide.md

```diff
@@ -0,0 +1,632 @@
+
+### creating the `desc:handoff-receive` certificate
+
+![](./handoffs-3party4.webm)
+
+The receiver doesn't just get the `desc:handoff-give` and pass it along, it generates its own certificate which includes the signed `desc:handoff-give`. The receiver does this to include some additional information only the receiver has that the exporter will need to securely perform the handoff. The certificate the receiver creates is called a `desc:handoff-receive` and looks like this:
+
+```
+<desc:handoff-receive receiving-session  ; The ID of the receiver <-> exporter session
+                      receiving-side     ; The public key of the receiver in the receiver <-> exporter session
+                      handoff-count      ; positive integer
+                      signed-give>       ; desc:sig-envolope containing a desc:handoff-give received from the gifter
+```
+
+We can see the receiver adds some information, but most of it is provided already by the gifter and is included by including that signed certificate along with the `desc:handoff-receive` in the `signed-give` field. One piece of information that's included here is the `handoff-count`, each session has an inrecementing integer starting at zero which reflects how many handoff-receive's they've sent in the session. They also include an integer which reflects how many handoff receive's they've received (with them being the exporter) within the session. This is used to prevent replay attacks, the exporter must only provide the reference once to prevent replay attacks. The last thing to note here is this is also wrapped in a `desc:sig-envolope` with a signature, but this signature is created with the receiver's private key in the gifter <-> receiver session. Again, this might seem a bit strange for two reasons:
+
+1. Again, we're sending the `desc:handoff-receive` to the exporter but using a key from another session, the gifter <-> receiver session.
+2. Since these keys are per-session and generated anew on each new connection, the exporter normally would have now way to verify this signature. The gifter, however, included the receiver's public key they used within the `handoff-give` (the `receiver-key` field) that they signed. The exporter is able to extract this key from the `desc:handoff-give` and use it to verify the signature on the `handoff-receive` (more on this later in the document).
```

Comment:
```suggestion
2. Since these keys are per-session and generated anew on each new connection, the exporter normally would have no way to verify this signature. The gifter, however, included the receiver's public key they used within the `handoff-give` (the `receiver-key` field) that they signed. The exporter is able to extract this key from the `desc:handoff-give` and use it to verify the signature on the `handoff-receive` (more on this later in the document).
```